applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -105,6 +105,24 @@ static void reboot_request_handler(struct k_work *work)
 	sys_reboot(SYS_REBOOT_WARM);
 }
 
+static inline u32_t flash_page_start(u32_t offset)
+{
+	return offset & (~(FLASH_PAGE_SIZE - 1));
+}
+
+static bool is_new_page_started(u32_t offset, size_t data_size)
+{
+	if (unlikely(offset == 0)) {
```

Comment:
unlikely not needed

Reply:
this branch is not needed btw
note that it is covering the case that will be problematic for any offset that starts at the beginning of a page

Reply:
we need to erase the page before any write
e.g. cur_offset is 2*4096, data size is 24
the result is `0x2000 != 0x2000` that is false while it should be true because we are starting a new page.
on the other hand when you are writing the last piece of data that cross the page you will erase the page you have just written into.

Reply:
It will simply erase new page when reaching end of previous page (may cause problems only when we are operating at the end of FLASH sector). Discussed offline.